contributors/devel/contributor-experience/github-magic.md

```diff
@@ -0,0 +1,101 @@
+# Kubernetes Github Magic
+This document has information that you are not going to find in any other document in the Kubernetes org repositories. The majority is considered a group of tips and tricks which cover common project/git practices, and you can see it as a TL;DR. They will be beneficial to your aha! moment.
+
+**Table of Contents**
+- [Communicating Effectively](#communicating-effectively-on-github)
+  - [How to be excellent to each other](#how-to-be-excellent-to-each-other)
+    - [Examples of good/bad communication](#examples-of-good/bad-communication)
+- [Submitting a contribution](#submitting-a-contribution)
+  - [Opening a issue](#opening-an-issue)
+  - [Opening a pull request](#opening-a-pull-request)
+    - [Pull requests advantages](#pull-requests-advantages)
+  - [Example of first comments when opening an issue or a PR](#example-of-first-comment-when-opening-an-issue-or-a-pr)
+  - [About trivial fixes](#about-trivial-fixes)
+  - [Working locally](#working-locally)
+    - [Branches](#branches)
+    - [Squashing Commits](#squashing-commits)
+
+## Communicating effectively on Github
+
+### How to be excellent to each other
+First, read the [Code of Conduct]
+
+#### Examples of good/bad communication
+  🙂 “X doesn’t compile when I do Y”  
+  😞 “X doesn’t work! Please fix it!”
+
+  When closing a PR we should convey an explanatory and cordial message talking about why it does not meet the requirements to merge it. It may offer feedback.  
+  **Example:**
+
+  🙂 “I’m closing this PR because this feature can’t support the use case X, but as you’ve propoused it can be a great idea to implement in Y tool. Thanks for  working on this.”
+
+  😞 “Why isn’t this following the api conventions? This is a complete failure!”
+
+## Submitting a Contribution
```

Comment:
For each of the sections, it might be worth linking to their associated section in the contributor/devel guide if the person reading the doc wants more information.

Reply:
Thank you. This is a good idea.